File: api/app/Models/Invoice/InvoiceCostgroupDistribution.php
<?php

namespace App\Models\Invoice;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\SoftDeletes;
use RichanFongdasen\EloquentBlameable\BlameableTrait;

class InvoiceCostgroupDistribution extends Model
{
    use SoftDeletes, BlameableTrait;

    protected $fillable = ['costgroup_number', 'invoice_id', 'weight'];

    public function invoice()
    {
        return $this->belongsTo(Invoice::class);
    }

    public function costgroup()
    {
        return $this->belongsTo(Costgroup::class);
    }
}

File: apir/app/models/invoice_cost_group_distribution.rb
# frozen_string_literal: true

class InvoiceCostGroupDistribution < ApplicationRecord
  belongs_to :costgroup
  belongs_to :invoice

  validates :costgroup, :invoice, :weight, presence: true
  validates :weight, numericality: { only_integer: true, greater_than_or_equal_to: 0 }
end
